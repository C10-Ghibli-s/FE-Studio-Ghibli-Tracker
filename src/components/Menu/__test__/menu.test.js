/**
 * @jest-environment jsdom
 */
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { Menu } from '../index';

describe('Menu Component', () => {
    test('Menu must be disabled when page renders', () => {
        render(<Menu/>);
        const menuDisabled = screen.queryByTestId("navBar");
        expect(menuDisabled).toBe(null);
    });
    test('Menu must render when the open Menu Button is clicked', () => {
        render(<Menu/>);
        fireEvent.click(screen.getByRole('button'));
        const menuRendered = screen.getByTestId("navBar");
        expect(menuRendered).toBeInTheDocument();
    })
    test('match snapshots with the links', () => {
        render(<Menu/>);
        fireEvent.click(screen.getByRole('button'));
        const menuRendered = screen.getByTestId("menu-Navbar");
        expect(menuRendered).toMatchSnapshot();
    })
});