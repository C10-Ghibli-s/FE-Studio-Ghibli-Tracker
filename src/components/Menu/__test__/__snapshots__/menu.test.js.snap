// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component match snapshots with the links 1`] = `
<div
  class="menuContainer"
  data-testid="menu-Navbar"
>
  <button
    class="menu__openMenuButton"
    data-testid="menuOpen-button"
    style="opacity: 0;"
    type="button"
  >
    <img
      src="test-file-stub"
    />
  </button>
  <nav>
    <ul
      data-testid="navBar"
      style="opacity: 0; transform: translateX(-20px) translateZ(0);"
    >
      <li
        style="opacity: 0; transform: translateY(-40px) translateZ(0);"
      >
        Settings
      </li>
      <li
        style="opacity: 0; transform: translateY(-40px) translateZ(0);"
      >
        <a
          href="/home"
        >
          home
        </a>
      </li>
      <li
        style="opacity: 0; transform: translateY(-40px) translateZ(0);"
      >
        <a
          href="/profile"
        >
          profile
        </a>
      </li>
      <li
        style="opacity: 0; transform: translateY(-40px) translateZ(0);"
      >
        <a
          href="/scores"
        >
          scores
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
